src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2504,6 +2520,10 @@ Node TheoryStringsRewriter::rewriteReplace( Node node ) {
             emptyNodes.insert(c[0]);
           }
         }
+        else
```

Comment:
Should we also set allEqs to false if we are EQUAL but neither side is empty?  Perhaps this should be allEmptyEqs?

Reply:
Yeah, good point. Done.